src/data/matrix/basic.lean

```diff
@@ -126,6 +124,8 @@ by simp [bit1_val, h]
 
 end numeral
 
+end diagonal
+
 @[simp] theorem diagonal_add [decidable_eq n] [add_monoid α] (d₁ d₂ : n → α) :
```

Comment:
Should we put this in the `diagonal` section while we're at it?

Reply:
Do you mean the `diagonal_add` theorem? That makes sense. The placement of the theorem outside of the section is likely a byproduct of other commits, since when it was created at https://github.com/leanprover-community/mathlib/commit/5260ab8e3669910516f3cbb8daa04d958a24856a, it was after the neg section. That's likely due to it being a theorem about addition rather than just construction.

Reply:
OK, I moved it up to be with the other `diagonal_*` lemmas at the same level of sectioning.